tests/basic/autopilot_test.go

```diff
@@ -1808,6 +1809,138 @@ func scheduleAppsWithAutopilot(testName string, testScaleFactor int, apRules []a
+			log.FailOnError(err, "Failed to get list of KVDB nodes from the cluster")
+			for _, kvdbNode := range kvdbNodes {
+				appNode, ok := stNodes[kvdbNode.ID]
+				if !ok {
+					log.InfoD("node with id %s not found in the nodes list", kvdbNode.ID)
+					continue
+				}
+				if storageNode.Name == appNode.Name {
+					//kvdb shouldn't be same as storagenodes
+					continue
+				}
+				crashedNodes = append(crashedNodes, appNode)
+				err := AddLabelsOnNode(appNode, poolLabel)
+				log.FailOnError(err, "Failed to label the node")
+				break
+			}
+		})
+
+		Step("schedule apps with autopilot rules for pool expand", func() {
+			err := AddLabelsOnNode(storageNode, poolLabel)
+			log.FailOnError(err, "Failed to add label to the node")
+			contexts = scheduleAppsWithAutopilot(testName, 1, apRules, scheduler.ScheduleOptions{PvcSize: 50 * units.GiB})
+		})
+
+		Step("Wait for Autopilot pool expansion, then crash a KVDB and Storage Node", func() {
+			//Crash one kvdb node and one storage node where the application is provisioned
+			log.InfoD("Started listening for any autopilot event")
+			err = aututils.WaitForAutopilotEvent(apRules[0], "", []string{aututils.AnyToTriggeredEvent})
+			log.FailOnError(err, "Failed to listen for any autopilot events")
+			err = aututils.WaitForAutopilotEvent(apRules[0], "", []string{aututils.ActiveActionsPendingToActiveActionsInProgress})
+			log.FailOnError(err, "Failed to listen for active-actions-pending to active-actions-in-progress event")
+			var wg sync.WaitGroup
+			for _, nodeToCrash := range crashedNodes {
+				wg.Add(1)
+
+				go func(nodeToCrash node.Node) {
+					defer wg.Done() // Decrement the WaitGroup counter when the task is done
+					stepLog := fmt.Sprintf("crash node: %s", nodeToCrash.Name)
+					Step(stepLog, func() {
+						log.InfoD(stepLog)
+						err := Inst().N.CrashNode(nodeToCrash, node.CrashNodeOpts{
+							Force: true,
+							ConnectionOpts: node.ConnectionOpts{
+								Timeout:         defaultCommandTimeout,
+								TimeBeforeRetry: defaultCommandRetry,
+							},
+						})
+						dash.VerifySafely(err, nil, "Validate node is crashed")
+					})
+
+					stepLog = fmt.Sprintf("wait for node: %s to be back up", nodeToCrash.Name)
+					Step(stepLog, func() {
+						log.InfoD(stepLog)
+						err := Inst().N.TestConnection(nodeToCrash, node.ConnectionOpts{
+							Timeout:         defaultTestConnectionTimeout,
+							TimeBeforeRetry: defaultWaitRebootRetry,
+						})
+						log.FailOnError(err, "Validate node is back up")
+					})
+
+					stepLog = fmt.Sprintf("wait for scheduler: %s and volume driver: %s to start",
+						Inst().S.String(), Inst().V.String())
+					Step(stepLog, func() {
+						log.InfoD(stepLog)
+						err := Inst().S.IsNodeReady(nodeToCrash)
+						log.FailOnError(err, "Validate node is ready")
+						err = Inst().V.WaitDriverUpOnNode(nodeToCrash, Inst().DriverStartTimeout)
+						log.FailOnError(err, "Validate volume driver is up")
```

Comment:
add validation that add-disk in completed successfully

Reply:
Done